distributed/comm/addressing.py

```diff
@@ -9,16 +9,21 @@
 DEFAULT_SCHEME = config.get('default-scheme', 'tcp')
 
 
-def parse_address(addr):
+def parse_address(addr, strict=False):
     """
     Split address into its scheme and scheme-dependent location string.
 
     >>> parse_address('tcp://127.0.0.1')
     ('tcp', '127.0.0.1')
+
+    If strict is set to true the address must have a scheme.
     """
     if not isinstance(addr, six.string_types):
         raise TypeError("expected str, got %r" % addr.__class__.__name__)
     scheme, sep, loc = addr.rpartition('://')
+    if strict and not sep:
+        raise ValueError("Invalid url scheme. Must be of the form <protocol>://<host>:<port>. "
+                        "Got %s" % addr)
```

Comment:
@sohaibiftikhar, the problem here is only the error message you adopted for the ValueError.